odl/space/fspace.py

```diff
@@ -359,192 +481,416 @@ def element(self, fcall=None, fapply=None):
             If fcall is a `FunctionSetVector`, it is wrapped
             as a new `FunctionSpaceVector`.
 
-        fapply : `callable`, optional
-            The actual instruction for in-place evaluation.
-            Its first argument must be the `FunctionSet.range` element
-            or the array of such (vectorization) to which the
-            result is written.
-
-            If fapply is a `FunctionSetVector`, it is wrapped
-            as a new `FunctionSpaceVector`.
+        vectorized : bool
+            Whether ``fcall`` supports vectorized evaluation.
 
         Returns
         -------
         element : `FunctionSpaceVector`
-            The new element.
+            The new element, always supports vectorization
+
+        Notes
+        -----
+        If you specify ``vectorized=False``, the function is decorated
+        with a vectorizer, which makes two elements created this way
+        from the same function being regarded as *not equal*.
         """
-        if fcall is None and fapply is None:
+        if fcall is None:
             return self.zero()
         else:
-            return super().element(fcall, fapply)
+            if not callable(fcall):
+                raise TypeError('function {!r} is not callable.'.format(fcall))
+            if not vectorized:
+                if self.field == RealNumbers():
```

Comment:
We have a nice little thing called [LinearSpace.element_type](https://github.com/odlgroup/odl/blob/967bc6e2253d20486b694e69c91f2e2cbc6c3adf/odl/set/space.py#L418-L421) which could perhaps be generalized?